v2/driver/driver.go

```diff
@@ -248,63 +252,126 @@ func resolveInputs(ctx context.Context, dag *metadata.DAG, task *pipelinespec.Pi
 		tasksCache = tasks
 		return tasks, nil
 	}
-	if task.GetInputs() != nil {
-		for name, paramSpec := range task.GetInputs().Parameters {
-			paramError := func(err error) error {
-				return fmt.Errorf("failed to resolve input parameter %s with spec %s: %w", name, paramSpec, err)
+	for name, paramSpec := range task.GetInputs().GetParameters() {
+		paramError := func(err error) error {
+			return fmt.Errorf("failed to resolve input parameter %s with spec %s: %w", name, paramSpec, err)
+		}
+		if paramSpec.GetParameterExpressionSelector() != "" {
+			return nil, paramError(fmt.Errorf("parameter expression selector not implemented yet"))
+		}
+		switch t := paramSpec.Kind.(type) {
+		case *pipelinespec.TaskInputsSpec_InputParameterSpec_ComponentInputParameter:
+			componentInput := paramSpec.GetComponentInputParameter()
```

Comment:
nit: compInputParam

Reply:
I personally feels that this is not necessary, because this whole section is about parameters, and the variable is used in this very small scope. It's unnecesary to include Param